src/psyclone/domain/common/algorithm/psyir.py

```diff
@@ -68,27 +73,35 @@ class AlgorithmInvokeCall(Call):
     _text_name = "AlgorithmInvokeCall"
     _colour = "green"
 
-    def __init__(self, invoke_routine_symbol, index, parent=None):
+    def __init__(self, invoke_routine_symbol, index, parent=None, name=None):
         super().__init__(invoke_routine_symbol, parent=parent)
 
         if not isinstance(index, int):
             raise TypeError(
-                "AlgorithmInvokeCall index argument should be an int but "
-                "found '{0}'.".format(type(index).__name__))
+                f"AlgorithmInvokeCall index argument should be an int but "
+                f"found '{type(index).__name__}'.")
         if index < 0:
             raise ValueError(
-                "AlgorithmInvokeCall index argument should be a non-negative "
-                "integer but found {0}.".format(index))
-
+                f"AlgorithmInvokeCall index argument should be a non-negative "
+                f"integer but found {index}.")
+        if name and not isinstance(name, str):
+            raise TypeError(
+                f"AlgorithmInvokeCall name argument should be a str but "
+                f"found '{type(name).__name__}'.")
         self._index = index
-        self.psylayer_routine_symbol = None
+        # Keep the root names as these will also be needed by the
+        # PSy-layer to use as tags to pull out the actual names from
+        # the algorithm symbol table, once issue #753 is complete.
+        self.psylayer_routine_root_name = None
+        self.psylayer_container_root_name = None
```

Comment:
Usually we have attributes with a _ prefix and a property getter to access them (do we have a rule that it always must be this way?). But in any case, it may be a good idea here regardless if the names should always be created following the rules of "create_psylayer_symbol_root_names" and not by a direct assignment.

Reply:
Yes good point. At the moment they are only used internally so there is no need to make accessors.